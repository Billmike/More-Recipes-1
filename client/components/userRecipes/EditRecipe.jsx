import React from 'react';
import PropTypes from 'prop-types';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';

const customContentStyle = {
  width: '35%',
  maxWidth: '35%',
};

const EditRecipe = (props) => {
  const actions = [
    <FlatButton
      label="Cancel"
      secondary
      onClick={props.handleClose}
    />
  ];

  return (
    <div>
      <Dialog
        title="Edit Recipe"
        actions={actions}
        modal
        contentStyle={customContentStyle}
        open={props.open}
        autoScrollBodyContent
      >
        <div id="edit-recipe" className="row">
          <form className="col s12">
            <div className="modal-content">
              <div className="row">
                <div className="input-field col s12">
                  <input id="recipe-name" type="text" className="validate" />
                  <label htmlFor="recipe-name">Title</label>
                </div>
              </div>
              <div className="row">
                <div className="input-field col s12">
                  <input
                    id="recipe-description"
                    className="materialize-textarea"
                  />
                  <label htmlFor="recipe-description">Description</label>
                </div>
              </div>
              <div className="row">
                <div className="input-field col s12">
                  <input id="recipe-prep-time" type="number" min="0" />
                  <label htmlFor="recipe-prep-time">
                    Preparation Time (MINS)
                  </label>
                </div>
              </div>
              <div className="row">
                <div className="input-field col s12">
                  <textarea
                    id="recipe-ingredients"
                    className="materialize-textarea"
                  />
                  <label htmlFor="recipe-ingredients">Ingredients</label>
                </div>
              </div>
              <div className="row">
                <div className="input-field col s12">
                  <textarea
                    id="recipe-directions"
                    className="materialize-textarea"
                  />
                  <label htmlFor="recipe-directions">Directions</label>
                </div>
              </div>
            </div>
            <div className="modal-footer">
              <button
                type="submit"
                className={`col s6 offset-s3 btn btn-large waves-effect
                  waves-light indigo accent-2`}
              >
                Save
              </button>
            </div>
          </form>
        </div>
      </Dialog>
    </div>
  );
};

EditRecipe.propTypes = {
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
};

export default EditRecipe;
