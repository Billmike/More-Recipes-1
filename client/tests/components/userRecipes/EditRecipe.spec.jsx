import React from 'react';

import EditRecipe from '../../../components/userRecipes/EditRecipe';

describe('EditRecipe component', () => {
  it('should render component', () => {
    const props = {
      open: true,
      handleClose: jest.fn(),
      handleEditChange: jest.fn(),
      handleSelect: jest.fn(),
      handleEditRecipe: jest.fn(),
      handleDrop: jest.fn(),
      imagePreview: '',
      recipeToEdit: {
        title: 'Beans',
        category: 'Main',
        description: 'Tasty beans',
        preparationTime: 85,
        ingredients: 'Beans, onions, palm oil, salt',
        directions: 'Do this. Do that',
      },
      userRecipes: {
        isLoading: false,
        imageUploading: false,
        error: {},
      },
    };
    const wrapper = shallow(<EditRecipe {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should show spinner when updating recipe', () => {
    const props = {
      open: true,
      handleClose: jest.fn(),
      handleEditChange: jest.fn(),
      handleSelect: jest.fn(),
      handleEditRecipe: jest.fn(),
      handleDrop: jest.fn(),
      imagePreview: 'Recipe image',
      recipeToEdit: {
        title: 'Beans',
        category: 'Main',
        description: 'Tasty beans',
        preparationTime: 85,
        ingredients: 'Beans, onions, palm oil, salt',
        directions: 'Do this. Do that',
      },
      userRecipes: {
        isLoading: true,
        imageUploading: true,
        error: {},
      },
    };
    const wrapper = shallow(<EditRecipe {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show error message if any', () => {
    const props = {
      open: true,
      handleClose: jest.fn(),
      handleEditChange: jest.fn(),
      handleSelect: jest.fn(),
      handleEditRecipe: jest.fn(),
      handleDrop: jest.fn(),
      imagePreview: 'Recipe image',
      recipeToEdit: {
        title: 'Beans',
        category: 'Main',
        description: 'Tasty beans',
        preparationTime: 85,
        ingredients: 'Beans, onions, palm oil, salt',
        directions: 'Do this. Do that',
      },
      userRecipes: {
        isLoading: false,
        imageUploading: false,
        error: {
          message: 'Error updating recipe',
        },
      },
    };
    const wrapper = shallow(<EditRecipe {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('ErrorMessage').dive().text())
      .toEqual('Error updating recipe');
    expect(wrapper).toMatchSnapshot();
  });
});
