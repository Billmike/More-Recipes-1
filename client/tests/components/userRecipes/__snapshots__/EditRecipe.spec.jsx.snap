// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipe component should render component 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          id="cancel-edit-button"
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Edit Recipe"
  >
    <div
      className="row"
      id="edit-recipe"
    >
      <form
        className="col s12"
        onSubmit={[MockFunction]}
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              defaultValue="Beans"
              id="title"
              maxLength="255"
              name="title"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <label
              className="active"
              htmlFor="title"
            >
              Title
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <select
            className="browser-default"
            defaultValue="Main"
            onChange={[MockFunction]}
          >
            <option
              disabled={true}
              value=""
            >
              Select Category
            </option>
            <option
              value="Breakfast"
            >
              Breakfast
            </option>
            <option
              name="Lunch"
              value="Lunch"
            >
              Lunch
            </option>
            <option
              value="Dinner"
            >
              Dinner
            </option>
            <option
              value="Appetizer"
            >
              Appetizer
            </option>
            <option
              value="Main"
            >
              Main
            </option>
            <option
              value="Dessert"
            >
              Dessert
            </option>
          </select>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Tasty beans"
              id="description"
              maxLength="255"
              name="description"
              onChange={[MockFunction]}
              required={true}
            />
            <label
              className="active"
              htmlFor="description"
            >
              Description
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              defaultValue={85}
              id="preparationTime"
              min="0"
              name="preparationTime"
              onChange={[MockFunction]}
              required={true}
              type="number"
            />
            <label
              className="active"
              htmlFor="preparationTime"
            >
              Preparation Time (MINS)
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Beans, onions, palm oil, salt"
              id="ingredients"
              name="ingredients"
              onChange={[MockFunction]}
              placeholder="Enter ingredients separated by comma"
              required={true}
            />
            <label
              className="active"
              htmlFor="ingredients"
            >
              Ingredients
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Do this. Do that"
              id="directions"
              name="directions"
              onChange={[MockFunction]}
              placeholder="Enter directions separated by full stop"
              required={true}
            />
            <label
              className="active"
              htmlFor="directions"
            >
              Directions
            </label>
          </div>
        </div>
        <div
          className="row center-align"
        >
          <div
            className="col s12"
          >
            <div
              className="dropzone-area"
            >
              <Dropzone
                accept="image/*"
                className="dropzone"
                disableClick={false}
                disablePreview={false}
                disabled={false}
                maxSize={Infinity}
                minSize={0}
                multiple={false}
                onDrop={[MockFunction]}
                preventDropOnDocument={true}
              >
                <p
                  className="center-align"
                >
                  Drop your file or click here to upload
                  <i
                    className="material-icons center large"
                  >
                    camera
                  </i>
                </p>
              </Dropzone>
              <div
                className="row"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                    indigo accent-2"
            disabled={false}
            name="edit-save"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;

exports[`EditRecipe component should show error message if any 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          id="cancel-edit-button"
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Edit Recipe"
  >
    <div
      className="row"
      id="edit-recipe"
    >
      <form
        className="col s12"
        onSubmit={[MockFunction]}
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              defaultValue="Beans"
              id="title"
              maxLength="255"
              name="title"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <label
              className="active"
              htmlFor="title"
            >
              Title
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <select
            className="browser-default"
            defaultValue="Main"
            onChange={[MockFunction]}
          >
            <option
              disabled={true}
              value=""
            >
              Select Category
            </option>
            <option
              value="Breakfast"
            >
              Breakfast
            </option>
            <option
              name="Lunch"
              value="Lunch"
            >
              Lunch
            </option>
            <option
              value="Dinner"
            >
              Dinner
            </option>
            <option
              value="Appetizer"
            >
              Appetizer
            </option>
            <option
              value="Main"
            >
              Main
            </option>
            <option
              value="Dessert"
            >
              Dessert
            </option>
          </select>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Tasty beans"
              id="description"
              maxLength="255"
              name="description"
              onChange={[MockFunction]}
              required={true}
            />
            <label
              className="active"
              htmlFor="description"
            >
              Description
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              defaultValue={85}
              id="preparationTime"
              min="0"
              name="preparationTime"
              onChange={[MockFunction]}
              required={true}
              type="number"
            />
            <label
              className="active"
              htmlFor="preparationTime"
            >
              Preparation Time (MINS)
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Beans, onions, palm oil, salt"
              id="ingredients"
              name="ingredients"
              onChange={[MockFunction]}
              placeholder="Enter ingredients separated by comma"
              required={true}
            />
            <label
              className="active"
              htmlFor="ingredients"
            >
              Ingredients
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Do this. Do that"
              id="directions"
              name="directions"
              onChange={[MockFunction]}
              placeholder="Enter directions separated by full stop"
              required={true}
            />
            <label
              className="active"
              htmlFor="directions"
            >
              Directions
            </label>
          </div>
        </div>
        <div
          className="row center-align"
        >
          <div
            className="col s12"
          >
            <div
              className="dropzone-area"
            >
              <Dropzone
                accept="image/*"
                className="dropzone"
                disableClick={false}
                disablePreview={false}
                disabled={false}
                maxSize={Infinity}
                minSize={0}
                multiple={false}
                onDrop={[MockFunction]}
                preventDropOnDocument={true}
              >
                <img
                  alt=""
                  src="Recipe image"
                />
              </Dropzone>
              <div
                className="row"
              />
            </div>
          </div>
        </div>
        <ErrorMessage
          message="Error updating recipe"
        />
        <div
          className="row"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                    indigo accent-2"
            disabled={false}
            name="edit-save"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;

exports[`EditRecipe component should show spinner when updating recipe 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          id="cancel-edit-button"
          label="Cancel"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[MockFunction]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={false}
          secondary={true}
        />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={true}
    modal={true}
    open={true}
    repositionOnUpdate={true}
    title="Edit Recipe"
  >
    <div
      className="row"
      id="edit-recipe"
    >
      <form
        className="col s12"
        onSubmit={[MockFunction]}
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              defaultValue="Beans"
              id="title"
              maxLength="255"
              name="title"
              onChange={[MockFunction]}
              required={true}
              type="text"
            />
            <label
              className="active"
              htmlFor="title"
            >
              Title
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <select
            className="browser-default"
            defaultValue="Main"
            onChange={[MockFunction]}
          >
            <option
              disabled={true}
              value=""
            >
              Select Category
            </option>
            <option
              value="Breakfast"
            >
              Breakfast
            </option>
            <option
              name="Lunch"
              value="Lunch"
            >
              Lunch
            </option>
            <option
              value="Dinner"
            >
              Dinner
            </option>
            <option
              value="Appetizer"
            >
              Appetizer
            </option>
            <option
              value="Main"
            >
              Main
            </option>
            <option
              value="Dessert"
            >
              Dessert
            </option>
          </select>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Tasty beans"
              id="description"
              maxLength="255"
              name="description"
              onChange={[MockFunction]}
              required={true}
            />
            <label
              className="active"
              htmlFor="description"
            >
              Description
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <input
              className="validate"
              defaultValue={85}
              id="preparationTime"
              min="0"
              name="preparationTime"
              onChange={[MockFunction]}
              required={true}
              type="number"
            />
            <label
              className="active"
              htmlFor="preparationTime"
            >
              Preparation Time (MINS)
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Beans, onions, palm oil, salt"
              id="ingredients"
              name="ingredients"
              onChange={[MockFunction]}
              placeholder="Enter ingredients separated by comma"
              required={true}
            />
            <label
              className="active"
              htmlFor="ingredients"
            >
              Ingredients
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <textarea
              className="materialize-textarea validate"
              defaultValue="Do this. Do that"
              id="directions"
              name="directions"
              onChange={[MockFunction]}
              placeholder="Enter directions separated by full stop"
              required={true}
            />
            <label
              className="active"
              htmlFor="directions"
            >
              Directions
            </label>
          </div>
        </div>
        <div
          className="row center-align"
        >
          <div
            className="col s12"
          >
            <div
              className="dropzone-area"
            >
              <Dropzone
                accept="image/*"
                className="dropzone"
                disableClick={false}
                disablePreview={false}
                disabled={false}
                maxSize={Infinity}
                minSize={0}
                multiple={false}
                onDrop={[MockFunction]}
                preventDropOnDocument={true}
              >
                <img
                  alt=""
                  src="Recipe image"
                />
              </Dropzone>
              <div
                className="row"
              />
              <Spinner
                size="small"
              />
            </div>
          </div>
        </div>
        <div
          className="row"
        />
        <div
          className="row center-align"
        >
          <button
            className="btn btn-large waves-effect waves-light
                    indigo accent-2"
            disabled={true}
            name="edit-save"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </Dialog>
</div>
`;
